### PR TITLE
:bug: Render dependency versions as links

### PR DESCRIPTION
**Pull request by sjd78**
Resolves: #1340

On the Dependencies page, detail drawer, applications table: render the version text as a link to the maven central repository.  The link uses the dependency's sha as the key in the maven central search.

Add component `ExternalLink` to standardize rendering links outside of the app opening in a new tab.

Note: The resolution to #1338 will require a change to the application table's query filter and response object.  This will be done in PR #1536.  See https://github.com/konveyor/tackle2-hub/issues/557 for the response object change.

Screenshot:
image

### CODE REVIEW

File: client/src/app/utils/utils.ts [hunk 1]
```diff
@@ -151,3 +151,21 @@ export const formatPath = (path: Paths, data: any) => {
...
+const SHA_REGEX =
+  /([a-f0-9]{128}|[a-f0-9]{96}|[a-f0-9]{64}|[a-f0-9]{56}|[a-f0-9]{40})/g;
+
+/**
+ * In any given string, find the first thing that looks like a sha hash and return it.
+ * If nothing looks like a sha hash, return undefined.
+ */
+export const extractFirstSha = (str: string): string | undefined => {
+  const match = str.match(SHA_REGEX);
+  console.log("match:", match);
```
Comment (ibolton336) [inline client/src/app/utils/utils.ts:169]: Spare log.

File: client/src/app/pages/dependencies/dependency-apps-table.tsx [hunk 2]
```diff
@@ -229,3 +237,32 @@ export const DependencyAppsTable: React.FC<IDependencyAppsTableProps> = ({
...
   );
 };
+
+const DependencyVersionColumn = ({
+  appDependency: {
+    dependency: {
+      provider,
+      name,
+      version,
+      rule: sha, // TODO: rename to 'sha' with https://github.com/konveyor/tackle2-hub/issues/557
```
Comment (ibolton336) [inline client/src/app/pages/dependencies/dependency-apps-table.tsx:247]: LGTM - wondering if we should hold off for the hub PR to merge?

Comment (sjd78) [reply client/src/app/pages/dependencies/dependency-apps-table.tsx:247]: Option 1 - hold off and update this PR at the same time

Option 2 - merge this PR and fix the payload with #1536 

I'm leaning towards option 2, but it is not a strong preference.

Comment (ibolton336) [reply client/src/app/pages/dependencies/dependency-apps-table.tsx:247]: Merging and circling back sounds ok to me.

**Review by ibolton336 — APPROVED** [review]
Tested and LGTM

**Comment by sjd78** [comment]
Merging blocked, waiting on:
  - hub for https://github.com/konveyor/tackle2-hub/pull/558 (for a field name change)
  - addon-analyzer for https://github.com/konveyor/tackle2-addon-analyzer/pull/62 (to see if the dependency provider or labels will contain any language identifying text so the link is only generated for java dependencies)

**Pull request merged**